File: planning-idsdp-pps-frontend/src/app/main/modules/dpp-tapp-management/services/tapp-project-details.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { CrudRequestService } from '../../../core/services/crud-request.service';
import { environment } from '../../../../../environments/environment';
import { TappProjectDetailsModel } from '../models/tappProjectDetails.model';
import { Observable } from 'rxjs';
import {map} from "rxjs/operators";

@Injectable({
    providedIn: 'root',
})
export class TappProjectDetailsService extends CrudRequestService<TappProjectDetailsModel> {

    constructor(private http: HttpClient, ) {
      super(http, environment.ibcs.ppsPcBaseEndPoint+ 'project-details/');
  }

    getData(route): Observable<any> {
        const url = this._BASE_URL + route;
        return this.http.get(url);
    }

    saveData(model: any) {
        const url: string = environment.ibcs.ppsDppBackendPoint + 'project-details/create';
        return this.http.post(url, model).pipe(map((res: any) => {
            return res;
        }));
    }

    updateData(model: any, pcUuid: any) {
        const url: string = environment.ibcs.ppsDppBackendPoint + 'project-details/update-data'+ '/' + pcUuid;
        return this.http.put(url, model).pipe(map((res: any) => {
            return res;
        }));
    }

    getProjectDetails(pcUuid){
        const url: string = environment.ibcs.ppsDppBackendPoint + 'project-details/getProjectDetails'+'/'+pcUuid;
        return this.http.get(url).pipe(map((res: any) => {
            return res;
        }));
    }
}

File: planning-idsdp-pps-dpp-backend/src/main/java/com/ibcs/idsdp/dpptapp/services/TappProjectDetailsService.java
package com.ibcs.idsdp.dpptapp.services;

import com.ibcs.idsdp.common.web.dto.response.ResponseStatus;
import com.ibcs.idsdp.common.web.dto.response.ResponseWithResults;
import com.ibcs.idsdp.dpptapp.model.domain.TappProjectDetails;
import com.ibcs.idsdp.dpptapp.web.dto.request.TappProjectDetailsRequest;

public interface TappProjectDetailsService {

    /**
     * For create tapp project details records
     * @param tappProjectDetailsRequest
     * @return
     */
    ResponseStatus save(TappProjectDetailsRequest tappProjectDetailsRequest);

    /**
     * For get tapp project details records
     * @return
     */
    TappProjectDetails getFirstRowData ();

    ResponseWithResults getProjectDetailsData(String pcUuid);

    ResponseStatus updateProjectDetails(TappProjectDetailsRequest request, String pcUuid);
}
